salt/apache/cove.conf

```diff
@@ -1,29 +1,49 @@
 # vi: ft=apache
 
-<VirtualHost *:80>
-        ServerName {% if branch != pillar.default_branch %}{{ branch }}.{% endif %}{{ pillar.domain_prefix }}cove.opendataservices.coop
-        ServerAdmin code@opendataservices.coop
-        ServerAlias cove.opendataservices.coop
-        ServerAlias  cove.*.default.opendataservices.uk0.bigv.io
-        ServerAlias *.standard.open-contracting.org standard.open-contracting.org
-        DocumentRoot {{ djangodir }} 
+{% if https == 'yes' or https == 'force' %}
+{% set myportlist = [ '80', '443' ] %}
+{% else %}
+{% set myportlist = [ '80' ] %}
+{% endif %}
 
-        ErrorDocument 500 "<h2>Sorry, something went wrong.</h2> <p>Sometimes this happens because the input file is too big - maybe try again with a smaller sample.</p><p>Please file a <a href=\"https://github.com/OpenDataServices/cove/issues/new\">GitHub issue</a> or email <a href=\"mailto:code@opendataservices.coop\">code@opendataservices.coop</a> if this problem persists.</p>"
+{% for myport in myportlist %}
+
+{% if myport == '443' %}
+<IfModule mod_ssl.c>
+{% set myhttprotocol='https' %}
+{% else %}
+{% set myhttprotocol='http' %}
+{% endif %}
+<VirtualHost *:{{ myport }}>
+
+    ServerName {{ servername }}
+    {% for myalias in serveraliases %}
+    ServerAlias {{ myalias }}
+    {% endfor %}
+    ServerAdmin code@opendataservices.coop
+    {% if myport=='80' and https=='force' %}
+        Redirect / https://{{ servername }}
+    {% else %}
+
+        {% if myport=='443' %}
+        Include /etc/letsencrypt/options-ssl-apache.conf
+        SSLCertificateFile      /etc/letsencrypt/live/{{ servername }}/cert.pem
+        SSLCertificateKeyFile   /etc/letsencrypt/live/{{ servername }}/privkey.pem
+        SSLCertificateChainFile /etc/letsencrypt/live/{{ servername }}/chain.pem
+        {% endif %}
+
+        {# START OF SERVICE CONFIG #}
 
-        RemoteIPHeader X-Forwarded-For
-        # Truest Proxies:
-        # tmp2
-        RemoteIPTrustedProxy 46.43.3.132
-        RemoteIPTrustedProxy 2001:41c9:1:420::132
-        # live2
-        RemoteIPTrustedProxy 46.43.2.168
-        RemoteIPTrustedProxy 2001:41c9:1:41f::168 
```

Comment:
f***
that's a cut-and-paste error, in fact, I specifically remember thinking 'i hope i don't lose that buffer'
well spotted, thanky muchy